Homework

Dustingyd opened the pull request:
### 434
```python
class Solution: #创建Solution类
    def countSegments(self, s: str) -> int: #定义了一个名为 countSegments 的函数，该方法接收一个参数 s，这个参数是一个字符串（str类型），并且函数返回一个整数（int类型）。
        return len(s.split()) # Python split() 通过指定分隔符对字符串进行切片，如果参数 num 有指定值，则分隔 num+1 个子字符串，默认为所有的空字符，包括空格、换行(\n)、制表符(\t)等。
        # Python len() 方法返回对象（字符、列表、元组等）长度或项目个数。
```  
### 1869
```python
class Solution:
    def checkZeroOnes(self, s: str) -> bool: #定义了一个名为 checkZeroOnes 的函数，该方法接收一个参数 s，这个参数是一个字符串（str类型），并且函数返回 bool 型数据。
        Length_0 = [] # 初始化存放连续0长度的 list   
        Length_1 = [] # 初始化存放连续1长度的 list 
        for i in s.split('0'): # 按“0”分割
            Length_1.append(len(i)) # 分割完统计每个连续1区域的长度
        for j in s.split('1'): # 按“1”分割
            Length_0.append(len(j)) # 分割完统计每个连续0区域的长度
        return max(Length_1) > max(Length_0)
```
### 1784
```python
class Solution:
    def checkOnesSegment(self, s: str) -> bool:
        sp = s.split("0") # 按 0 分割 
        while '' in sp: # 循环删掉空段
            sp.remove('') # 删掉空段
        return len(sp) < 2 #判断非空段个数是否为0或1
```
### 852
```python
class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
        peak = max(arr) # 寻找山峰
        return arr.index(peak) # 返回山峰的位置
```
### 162
```python
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        peak = max(nums) # 寻找最高的山峰
        return nums.index(peak) # 返回最高的山峰，即为其中一个峰值元素的位置
```

iphysresearch commented:
Good Job!!!   @Dustingyd
---
Python homework:
- Total questions: 108
- Correct answers: 105
- Score: 97.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

Dustingyd commented:
### 434
```python
class Solution: #创建Solution类
    def countSegments(self, s: str) -> int: #定义了一个名为 countSegments 的函数，该方法接收一个参数 s，这个参数是一个字符串（str类型），并且函数返回一个整数（int类型）。
        return len(s.split()) # Python split() 通过指定分隔符对字符串进行切片，如果参数 num 有指定值，则分隔 num+1 个子字符串，默认为所有的空字符，包括空格、换行(\n)、制表符(\t)等。
        # Python len() 方法返回对象（字符、列表、元组等）长度或项目个数。
```  
### 1869
```python
class Solution:
    def checkZeroOnes(self, s: str) -> bool: #定义了一个名为 checkZeroOnes 的函数，该方法接收一个参数 s，这个参数是一个字符串（str类型），并且函数返回 bool 型数据。
        Length_0 = [] # 初始化存放连续0长度的 list   
        Length_1 = [] # 初始化存放连续1长度的 list 
        for i in s.split('0'): # 按“0”分割
            Length_1.append(len(i)) # 分割完统计每个连续1区域的长度
        for j in s.split('1'): # 按“1”分割
            Length_0.append(len(j)) # 分割完统计每个连续0区域的长度
        return max(Length_1) > max(Length_0)
```
### 1784
```python
class Solution:
    def checkOnesSegment(self, s: str) -> bool:
        sp = s.split("0") # 按 0 分割 
        while '' in sp: # 循环删掉空段
            sp.remove('') # 删掉空段
        return len(sp) < 2 #判断非空段个数是否为0或1
```
### 852
```python
class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
        peak = max(arr) # 寻找山峰
        return arr.index(peak) # 返回山峰的位置
```
### 162
```python
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        peak = max(nums) # 寻找最高的山峰
        return nums.index(peak) # 返回最高的山峰，即为其中一个峰值元素的位置
```

iphysresearch commented:
Good Job!!!   @Dustingyd
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

Dustingyd commented:
@iphysresearch 王老师好！我的作业完成（题目里面好多坑哈哈哈），您辛苦了！

Dustingyd commented:
@iphysresearch 王老师好！我的作业完成了（题目里好多陷阱），您辛苦了！！！

iphysresearch commented:
> @iphysresearch 王老师好！我的作业完成了（题目里好多陷阱），您辛苦了！！！

很棒哈！把扩展作业也做一下吧哈？

Dustingyd commented:
### 434
```python
class Solution: #创建Solution类
    def countSegments(self, s: str) -> int: #定义了一个名为 countSegments 的函数，该方法接收一个参数 s，这个参数是一个字符串（str类型），并且函数返回一个整数（int类型）。
        return len(s.split()) # Python split() 通过指定分隔符对字符串进行切片，如果参数 num 有指定值，则分隔 num+1 个子字符串，默认为所有的空字符，包括空格、换行(\n)、制表符(\t)等。
        # Python len() 方法返回对象（字符、列表、元组等）长度或项目个数。
```  
### 1869
```python
class Solution:
    def checkZeroOnes(self, s: str) -> bool: #定义了一个名为 checkZeroOnes 的函数，该方法接收一个参数 s，这个参数是一个字符串（str类型），并且函数返回 bool 型数据。
        Length_0 = [] # 初始化存放连续0长度的 list   
        Length_1 = [] # 初始化存放连续1长度的 list 
        for i in s.split('0'): # 按“0”分割
            Length_1.append(len(i)) # 分割完统计每个连续1区域的长度
        for j in s.split('1'): # 按“1”分割
            Length_0.append(len(j)) # 分割完统计每个连续0区域的长度
        return max(Length_1) > max(Length_0)
```
### 1784
```python
class Solution:
    def checkOnesSegment(self, s: str) -> bool:
        sp = s.split("0") # 按 0 分割 
        while '' in sp: # 循环删掉空段
            sp.remove('') # 删掉空段
        return len(sp) < 2 #判断非空段个数是否为0或1
```
### 852
```python
class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
        peak = max(arr) # 寻找山峰
        return arr.index(peak) # 返回山峰的位置
```
### 162
```python
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        peak = max(nums) # 寻找最高的山峰
        return nums.index(peak) # 返回最高的山峰，即为其中一个峰值元素的位置
```

Dustingyd commented:
> > @iphysresearch 王老师好！我的作业完成了（题目里好多陷阱），您辛苦了！！！
> 
> @Dustingyd 很棒哈！把扩展作业也做一下吧哈？

王老师好！扩展作业我之前 comment 了。现在我又重新 comment 了一遍，您看能看到吗？

iphysresearch commented:
哦哦 我看漏了哈！没问题了

Dustingyd commented:
> 哦哦 我看漏了哈！没问题了

谢谢老师，辛苦了！！
